al5d_gazebo/scripts/arm_controller.py

```diff
@@ -153,17 +170,48 @@ def loop(self):
         self.pos_pubs = []
         self.gripper_pubs = []
         for i in range(self.num_joints):
-            pub = rospy.Publisher("/al5d_arm_position_controller"+str(i+1)+"/command",
+            pub = rospy.Publisher(self.namespace+"al5d_arm_position_controller"+str(i+1)+"/command",
                                   Float64, queue_size=1, latch=True)
             self.pos_pubs.append(pub)
         for i in range(2):
-            pub = rospy.Publisher("/al5d_gripper_controller"+str(i+1)+"/command",
+            pub = rospy.Publisher(self.namespace+"al5d_gripper_controller"+str(i+1)+"/command",
                                   Float64, queue_size=1, latch=True)
             self.gripper_pubs.append(pub)
 
-        state_sub = rospy.Subscriber("/joint_states", JointState, self.state_cb)
-        pose_sub = rospy.Subscriber("/joint_poses", TransformStampedList, self.pose_cb)
-        contact_sub = rospy.Subscriber("/collision", ContactsState, self.contact_cb)
+        state_sub = rospy.Subscriber(self.namespace+"joint_states", JointState, self.state_cb)
+        pose_sub = rospy.Subscriber(self.namespace+"joint_poses", TransformStampedList, self.pose_cb)
+        contact_sub = rospy.Subscriber(self.namespace+"collision", ContactsState, self.contact_cb)
+
+
+        self.model_sub = rospy.Subscriber('/gazebo/model_states', ModelStates, self.model_cb);
+
+
+        for name, type in rospy.get_published_topics():
+            match = re.search('(\/.*\/)arm_interface\/state',name)
+            if match is not None:
+                name = match.group(1)
+                if name != self.namespace:
+                    self.opponent_sub = rospy.Subscriber(match.group(), JointState, self.opponent_cb);
+                    break
+
+
+
+            # % find all namespaced lynx arms in sim with any namespace
```

Comment:
delete large blocks of commented out code or explain why commented out.

Reply:
done